### ISSUE TITLE
[Bug]  T.copy 将3D BufferRegion(16,4,8) flatten为2D (4,8)（取最后2维度），而非总平面 (64,8)。批量DMA只拷了32元素（token 0），其余15个token UB未初始化

### ISSUE THREAD

**Issue by mottomottowardsworth-alt** [tile-ai/tilelang-ascend#1231]
## Bug Description
 copy_gm_to_ub 将3D BufferRegion(16,4,8) flatten为2D (4,8)（取最后2维度），而非总平面 (64,8)。批量DMA只拷了32元素（token 0），其余15个token UB未初始化

<img width="1756" height="206" alt="Image" src="https://github.com/user-attachments/assets/77cb26b0-1c96-4885-b1c2-c52640c172ac" />

## Environment

**Hardware:**
- NPU Device: [ A3]

**Software:**
- OS: [EulerOS]
- CANN version: [8.5.0]
- torch-npu version: [2.10.0RC2]
- Python version: [3.11]
- TileLang-Ascend version: [5a221177f184bc52e97620d06007b867af7bc95d]
- Installation method: [build from source]

## Code to Reproduce

```python
# Minimal reproducible code
import tilelang
import torch

FWD_PASS_CONFIGS = {
    tilelang.PassConfigKey.TL_ASCEND_AUTO_CV_COMBINE: True,
    tilelang.PassConfigKey.TL_ASCEND_AUTO_SYNC: False,
    tilelang.PassConfigKey.TL_ASCEND_MEMORY_PLANNING: True,
}

_BWD_PASS_CONFIGS = {
    tilelang.PassConfigKey.TL_ASCEND_AUTO_CV_COMBINE: True,
    tilelang.PassConfigKey.TL_ASCEND_AUTO_SYNC: True,
    tilelang.PassConfigKey.TL_ASCEND_MEMORY_PLANNING: True,
}

VEC_NUM = 2


@tilelang.jit(pass_configs=_FWD_PASS_CONFIGS)
def _mhc_sinkhorn_fwd(
    hidden_size: int,
    token_block_size: int,
    repeat: int,
    eps: float,
) -> tilelang.JITKernel:
    assert hidden_size == 4
    num_tokens = T.symbolic('num_tokens')
    dtype = "float32"
    pad_h = ((hidden_size + 7) // 8) * 8
    sub_block_tokens = token_block_size // VEC_NUM

    @T.prim_func
    def mhc_sinkhorn_kernel(
        comb_res_mix: T.Tensor[(num_tokens, hidden_size, pad_h), dtype],
        comb_res_mix_out: T.Tensor[(num_tokens, hidden_size, hidden_size), dtype],
    ) -> None:
        with T.Kernel(T.ceildiv(num_tokens, token_block_size), is_npu=True) as (cid, vid):
            comb_ub = T.alloc_ub((sub_block_tokens, hidden_size, pad_h), dtype)
            row_max_col = T.alloc_ub((hidden_size, 1), dtype)
            row_max_2d = T.alloc_ub((hidden_size, pad_h), dtype)
            row_sum_col = T.alloc_ub((hidden_size, 1), dtype)
            row_sum_2d = T.alloc_ub((hidden_size, pad_h), dtype)
            col_sum_ub = T.alloc_ub((pad_h,), dtype)
            col_sum_2d = T.alloc_ub((hidden_size, pad_h), dtype)

            row_start = cid * token_block_size + vid * sub_block_tokens

            with T.Scope("V"):
                T.set_flag("mte3", "mte2", 0)
                T.wait_flag("mte3", "mte2", 0)

                for i_tok in T.serial(sub_block_tokens):
                    token_idx = row_start + i_tok
                    if token_idx < num_tokens:
                        T.copy(
                            comb_res_mix[token_idx, :, :],
                            comb_ub[i_tok, :, :],
                        )

                T.set_flag("mte2", "v", 0)
                T.wait_flag("mte2", "v", 0)

                for i_tok in T.serial(sub_block_tokens):
                    token_idx = row_start + i_tok
                    if token_idx < num_tokens:
                        T.reduce_max(comb_ub[i_tok, :, :], row_max_col, dim=-1)
                        T.tile.broadcast(row_max_2d, row_max_col)
                        T.tile.sub(comb_ub[i_tok, :, :], comb_ub[i_tok, :, :], row_max_2d)
                        T.tile.exp(comb_ub[i_tok, :, :], comb_ub[i_tok, :, :])

                        T.reduce_sum(comb_ub[i_tok, :, :], row_sum_col, dim=-1)
                        T.tile.broadcast(row_sum_2d, row_sum_col)
                        T.tile.add(row_sum_2d, row_sum_2d, eps)
                        T.tile.div(comb_ub[i_tok, :, :], comb_ub[i_tok, :, :], row_sum_2d)

                        T.reduce_sum(comb_ub[i_tok, :, :], col_sum_ub, dim=0)
                        T.tile.add(col_sum_ub, col_sum_ub, eps)
                        T.tile.broadcast(col_sum_2d, col_sum_ub)
                        T.tile.div(comb_ub[i_tok, :, :], comb_ub[i_tok, :, :], col_sum_2d)

                        for _ in T.serial(repeat - 1):
                            T.reduce_sum(comb_ub[i_tok, :, :], row_sum_col, dim=-1)
                            T.tile.broadcast(row_sum_2d, row_sum_col)
                            T.tile.add(row_sum_2d, row_sum_2d, eps)
                            T.tile.div(comb_ub[i_tok, :, :], comb_ub[i_tok, :, :], row_sum_2d)

                            T.reduce_sum(comb_ub[i_tok, :, :], col_sum_ub, dim=0)
                            T.tile.add(col_sum_ub, col_sum_ub, eps)
                            T.tile.broadcast(col_sum_2d, col_sum_ub)
                            T.tile.div(comb_ub[i_tok, :, :], comb_ub[i_tok, :, :], col_sum_2d)

                T.set_flag("v", "mte3", 0)
                T.wait_flag("v", "mte3", 0)

                for i_tok in T.serial(sub_block_tokens):
                    token_idx = row_start + i_tok
                    if token_idx < num_tokens:
                        T.copy(comb_ub[i_tok, 0:hidden_size, 0:hidden_size], comb_res_mix_out[token_idx, 0:hidden_size, 0:hidden_size])

                T.set_flag("mte3", "mte2", 0)
                T.wait_flag("mte3", "mte2", 0)

    return mhc_sinkhorn_kernel
```

## Error Message

```
Paste the full error message here
```

## Actual Behavior
A clear and concise description of what actually happened.

## Additional Context

**Programming Mode:**
- [ ] Developer mode (automatic sync/buffer management)
- [ ] Expert mode (manual T.Scope/T.barrier_all)
- [ ] Mixed mode

**Backend:**
- [ ] Ascend C & PTO backend
- [ ] AscendNPU IR backend

**Pass Configs (if applicable):**
```python
pass_configs = {
    tilelang.PassConfigKey.TL_ASCEND_AUTO_SYNC: ...,
    tilelang.PassConfigKey.TL_ASCEND_MEMORY_PLANNING: ...,
    # ...
}
```

**Operator Type:**
[e.g. GEMM, Flash Attention, Softmax, Convolution, etc.]

**Additional Information:**
- Have you checked the [Programming Guide](../docs/TileLang-Ascend%20Programming%20Guide.md)?
- Have you searched existing issues?
- Any other context or screenshots?
